pkg/services/ngalert/api/lotex_am.go

```diff
@@ -35,14 +50,57 @@ func NewLotexAM(proxy *AlertingProxy, log log.Logger) *LotexAM {
 	}
 }
 
-func (am *LotexAM) RouteGetAMStatus(ctx *models.ReqContext) response.Response {
+func (am *LotexAM) withAMReq(
+	ctx *models.ReqContext,
+	method string,
+	endpoint string,
+	pathParams []string,
+	body io.Reader,
+	extractor func(*response.NormalResponse) (interface{}, error),
+	headers map[string]string,
+) response.Response {
+	ds, err := am.DataProxy.DataSourceCache.GetDatasource(ctx.ParamsInt64(":Recipient"), ctx.SignedInUser, ctx.SkipCache)
+	if err != nil {
+		if errors.Is(err, models.ErrDataSourceAccessDenied) {
+			return ErrResp(http.StatusForbidden, err, "Access denied to datasource")
+		}
+		if errors.Is(err, models.ErrDataSourceNotFound) {
+			return ErrResp(http.StatusNotFound, err, "Unable to find datasource")
+		}
+		return ErrResp(http.StatusInternalServerError, err, "Unable to load datasource meta data")
+	}
+
+	impl := ds.JsonData.Get("implementation").MustString(defaultImplementation)
+	implEndpoints, ok := endpoints[impl]
+	if !ok {
+		return ErrResp(http.StatusBadRequest, fmt.Errorf("unsupported Alert Manager implementation \"%s\"", impl), "")
+	}
+	endpointPath, ok := implEndpoints[endpoint]
```

Comment:
Shouldn't we forbid also POST requests for the upstream?

Reply:
It doesn't have a get/set config endpoint at all, but other endpoints work the same as with cortex. You can still post silences or alerts